spec/github_helper_spec.rb

```diff
@@ -47,6 +47,25 @@
     end
   end
 
+  describe 'github_client' do
+    let(:client) do
+      instance_double(
+        Octokit::Client,
+        user: instance_double('User', name: 'test'),
+        'auto_paginate=': nil
+      )
```

Comment:
Only potential issue I have with that is if we change the implementation detail in the future, and for example the implementation ends up calling more methods… won't this risk calling the real ones (and even potentially make real API/network calls)?

I'm not totally sure tbh as I don't remember how test doubles work in Ruby/rspec, if they fallback to the real object's implementation if a method is not mocked (which we'd want to avoid in this case), or do they fail the test (what we'd want here)?

Reply:
Yep – with verifying doubles the test will fail if we call anything else. And if we try to make a network call it'll fail the test as well, so I feel like we're pretty covered there?